src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy

```diff
@@ -1519,6 +1501,22 @@ class GenericPipeline extends Pipeline {
         }
     }
 
+    Boolean checkAuthorizedUser() {
+        def isAuthorizedUser = false
+        Integer causeID = this.build.identifySurfaceCause()
+        if (causeID == PipelineConstants.UPSTREAM_CAUSE_ID) {
+            causeID = this.build.identifyRootCause()
+        }
+        if (causeID == PipelineConstants.USERID_CAUSE_ID) {
+            isAuthorizedUser = true
+        } else if (causeID == PipelineConstants.BRANCHEVENT_CAUSE_ID || causeID == PipelineConstants.BRANCHINDEXING_CAUSE_ID) {
+            if (this.changeInfo.isPullRequest) {
+                isAuthorizedUser = isPRAuthorizedUser()
+            }
```

Comment:
Can we add an else statement here, if it's not pull request, we always allow it to pass through.

The difference is:
- only trusted user can create a branch in the repository
- anyone with anonymous permission can create pull request.

So it's safe to say if it's branch, from jenkins library point of view, we allow it pass. If the pipeline itself decides not run branch build, it can define and customize in its own pipeline to disable this.